pkgs/tools/misc/fontforge/default.nix

```diff
@@ -9,22 +9,27 @@
 
 stdenv.mkDerivation rec {
   name = "fontforge-${version}";
-  version = "20160404";
+  version = "20170730";
 
   src = fetchFromGitHub {
     owner = "fontforge";
     repo = "fontforge";
     rev = version;
-    sha256 = "15nacq84n9gvlzp3slpmfrrbh57kfb6lbdlc46i7aqgci4qv6fg0";
+    sha256 = "15k6x97383p8l40jvcivalhwgbbcdg5vciyjz6m9r0lrlnjqkv99";
   };
 
-  patches = [(fetchpatch {
-    name = "use-system-uthash.patch";
-    url = "http://pkgs.fedoraproject.org/cgit/fontforge.git/plain/"
-      + "fontforge-20140813-use-system-uthash.patch?id=8bdf933";
-    sha256 = "0n8i62qv2ygfii535rzp09vvjx4qf9zp5qq7qirrbzm1l9gykcjy";
-  })];
-  patchFlags = "-p0";
+  patches = [
+    ./fontforge-20140813-use-system-uthash.patch
+
+    (fetchpatch {
+       url = "https://github.com/fontforge/fontforge/compare/${version}...volth:rb-${version}.patch";
```

Comment:
I am not sure I understood the question about reproducibility.
The problems with reproducibility were:
1. the current time leaked to the fonts
2. the fontforge compilation time leaked to the fonts
3. the current linux user name (```nixbld4```) leaked to the fonts
4. the mtime of source files leaked to the fonts (it is generally ok it they are unpacked from the archive, but sometimes the sources fed to fontforge are generated on the previous build step and its mtime is not detemenistic)
5. Difference between ```x87``` (default on ```i686-linux```, intermediate results are 80-bit) and ```sse``` (default on ```x86_64-linux```, intermediate results are 64-bit). It looks insignificant, but sometimes ```(a < b)``` is true on ```i686-linux``` and false on ```x86_64-linux``` because of the difference between ```a``` and ```b``` is so small that they are equal if the comparison is 64-bit and not equal if the comparison is 80-bit. ```pkgs.inconsolata-lgc``` and ```pkgs.libertine``` are examples when such 80/64-bit difference results in choosing different number of point to represent curves and very different font binaries